src/shared/particle_dynamics/particle_iterators.h

```diff
@@ -286,6 +348,52 @@ namespace SPH
 				return operation(x, y);
 			});
 	};
+
+    template <class ReturnType, typename Operation, class LocalDynamicsFunction, class Proxy>
+    inline ReturnType particle_reduce(const ParallelPolicy &par_policy, const size_t &all_real_particles,
+                                      ReturnType identity, Operation &operation,
```

Comment:
Operation should likely be a forwarding reference
```suggestion
                                      ReturnType identity, Operation &&operation,
```

Reply:
OK I see, it is used to access a type within the scope of Operation class/struct. This could be implemented externally without having to embed SYCLOp within each operation via (manual?) list of specialization `SYCLOp<Operation>::type`. The approach for passing operation in the standard library (and some other parallel framework iirc) is to pass the Operation without qualifiers (&approx; by value) (e.g. [`std::accumulate`](https://en.cppreference.com/w/cpp/algorithm/accumulate)).